HACKING.md

```diff
@@ -23,78 +23,98 @@ newgrp lxd
 Setup the environment by running:
 
 ```shell
-./tools/environment-setup.sh
+make setup
 ```
 
-To work inside this environment, run:
+### Tooling
+
+We use a large number of tools for our project. Most of these are installed for
+you with `make setup`, but you'll need to install Python 3.12 separately.
+
+### Testing
+
+For unit testing, you can use `make` recipes:
 
 ```shell
-lxc exec snapcraft-dev -- sudo -iu ubuntu bash
+make test  # All unit tests
+make test-fast  # Only the fast tests
+make test-slow  # Only the slow tests
 ```
 
-Import your keys (`ssh-import-id`) and add a `Host` entry to your ssh config if you are interested in [Code's](https://snapcraft.io/code) [Remote-SSH](https://code.visualstudio.com/docs/remote/ssh) plugin.
+For integration testing, Snapcraft uses [Spread](https://github.com/canonical/spread). Spread is a system for distributing tests and executing them in different backends, in parallel. In order to use spread, first build Snapcraft using:
 
-### Tooling
+```shell
+snapcraft pack
+```
 
-We use a large number of tools for our project. Most of these are installed for
-you with tox, but you'll need to install:
+Then, move the resulting snap into the tests directory:
 
-- Python 3.12 (default on Ubuntu 24.04) with setuptools.
-- [tox](https://tox.wiki) version 3.8 or later
-- [pyright](https://github.com/microsoft/pyright)  (also available via snap: `snap install pyright`)
-- [ruff](https://github.com/astral/ruff) (also available via snap: `snap install ruff`)
-- [ShellCheck](https://www.shellcheck.net/)  (also available via snap: `snap install shellcheck`)
+```shell
+mv *.snap tests/
+```
 
-### Testing
+Next, install Spread with Go:
+
+```
+go install github.com/snapcore/spread/cmd/spread@latest
+```
+
+Note: Ensure that the `$HOME/go/bin` directory is on your path after installation.
+
+Then, you can run the integration tests using a local LXD backend with:
+
+```
+spread -v lxd:
+```
 
-See the [Testing guide](TESTING.md).
+Or, you can run them in Google Cloud if you have a `SPREAD_GOOGLE_KEY`, with:
```

Comment:
I think we should provide a link to a doc that describes the key – it's not evident what this is or how to get it.

Reply:
Let's not do a link, then, but a quick phrase in that sentence describing it as Google Cloud credentials.